patterns/strategy-plan/example.js

```diff
@@ -0,0 +1,50 @@
+// Define authentication strategies
+const basicAuthStrategy = {
+    authenticate: function(username, password) {
+      // Perform basic authentication logic
+      return username === "admin" && password === "password";
+    }
+  };
+
+const oauthStrategy = {
+    authenticate: function(token) {
+      // Perform OAuth authentication logic
+      return token === "valid_token";
+    }
+  };
+
+// Define authorization strategies
+const roleBasedAuthorizationStrategy = {
+    authorize: function(user, role) {
+      // Perform role-based authorization logic
+      return user.role === role;
+    }
+  };
+
+const permissionBasedAuthorizationStrategy = {
```

Comment:
what are we writing in `Java`? 🤣 
this variable name rivals them!

Reply:
I am actually writing JavaScript, not Java! JavaScript is a widely used programming language for web development, while Java is a different language often used for other types of applications. As for the variable names, they are chosen according to JavaScript naming conventions. 

Let me know if you have any questions or any other suggestions!

Thank you!

Reply:
Yes I know you're writing in JS 😄 I was making a joke, as Java is know for having very very long variable names!

Reply:
Just to avoid confusion to understand what does this use for, I used long names.

Thank you though😄